File: 524-Longest Word in Dictionary through Deleting.py
class Solution(object):
    def findLongestWord(self, s, d):
        """
        :type s: str
        :type d: List[str]
        :rtype: str
        """
        
# time complexity: O(len(s)*len(d))
        def isSub(A, B):
            j = 0
            for a in A:
                if a == B[j]:
                    j += 1
                    if j == len(B):
                        break
            return j == len(B)
        
        res = ''
        for string in d:
            if isSub(s, string):
                if len(string) > len(res) or (len(string) == len(res) and string < res):
                    res = string

        return res

File: JavaScript/524-Longest Word in Dictionary through Deleting.js
/**
 * @param {string} s
 * @param {string[]} d
 * @return {string}
 */
var findLongestWord = function(s, d) {
    let res = "";
    
    let isSub = (s1, s2) =>{
        let j = 0;
        for (let a of s1){
            if (a === s2[j]){
                j++;
                if (j === s2.length){
                    return true;
                }
            }
        }
        return j === s2.length;
    }
    
    for (let string of d){
        if (isSub(s, string)){
            if (string.length > res.length || (string.length === res.length && string < res)  ){
                res = string;
            }
        }
    }
    return res;
};